### PR TITLE
Fix ACL for default case

### PR DESCRIPTION
**Pull request by RedbackThomson**
The Bucket ACL was not being reset back to the default value of `private` when removing the `ACL` value. This pull request ensures that if the user does not specify an ACL, the reconciler does not get caught in a loop trying to update the default grant header (grant full control for the current user) and does not try to set an empty string ACL.

### CODE REVIEW

File: pkg/resource/bucket/hook.go [hunk 1]
```diff
@@ -276,7 +277,42 @@ func customPreCompare(
...
+		}
+		if a.ko.Spec.GrantReadACP == nil && b.ko.Spec.GrantReadACP != nil {
+			a.ko.Spec.GrantReadACP = &emptyGrant
+		}
+		if a.ko.Spec.GrantWrite == nil && b.ko.Spec.GrantWrite != nil {
+			a.ko.Spec.GrantWrite = &emptyGrant
+		}
+		if a.ko.Spec.GrantWriteACP == nil && b.ko.Spec.GrantWriteACP != nil {
+			a.ko.Spec.GrantWriteACP = &emptyGrant
+		}
```
Comment (jaypipes) [inline pkg/resource/bucket/hook.go:313]: we might consider placing the above into a little utility function, ala `matchPossibleCannedACL`. But you can handle that in a future PR if you want...

**Review by jaypipes — APPROVED** [review]
:+1:

**Review by A-Hilaly — APPROVED** [review]

File: pkg/resource/bucket/hook.go (hunk 1 above)
Comment (RedbackThomson) [reply pkg/resource/bucket/hook.go:313]: I'm still not certain my logic here is flawless. Too many code paths to mentally track for me to be confident in it. I probably need to add some integration tests that switch between the types.

**Pull request merged**